### PR TITLE
Added Migration service for migrating old property to new property

### PR DESCRIPTION
**Pull request by rohit-eGov**

### CODE REVIEW

File: property-services/src/main/java/org/egov/pt/service/PropertyService.java [hunk 1]
```diff
@@ -232,4 +236,281 @@ private Boolean enrichCriteriaFromUser(PropertyCriteria criteria, RequestInfo re
...
+
+		List<Property> properties = new ArrayList<>();
+		for(OldProperty oldProperty : oldProperties){
+			Property property = new Property();
+			property.setId(UUID.randomUUID().toString());
+			property.setPropertyId(oldProperty.getPropertyId());
+			//property.setSurveyId();
+			property.setLinkedProperties(null);
+			property.setTenantId(oldProperty.getTenantId());
+			property.setAccountId(requestInfo.getUserInfo().getUuid());
```
Comment (talele08) [inline property-services/src/main/java/org/egov/pt/service/PropertyService.java]: should be taken from oldProperty

File: property-services/src/main/java/org/egov/pt/service/PropertyService.java [hunk 2]
```diff
@@ -232,4 +236,281 @@ private Boolean enrichCriteriaFromUser(PropertyCriteria criteria, RequestInfo re
...
+			property.setOldPropertyId(oldProperty.getOldPropertyId());
+			property.setStatus(Status.fromValue(oldProperty.getStatus().toString()));
+
+			if(oldProperty.getAddress() == null)
+				property.setAddress(null);
+			else
+				property.setAddress(migrateAddress(oldProperty.getAddress()));
+
+			property.setAcknowldgementNumber(oldProperty.getAcknowldgementNumber());
+			property.setPropertyType(oldProperty.getPropertyDetails().get(0).getPropertyType());
```
Comment (talele08) [inline property-services/src/main/java/org/egov/pt/service/PropertyService.java]: get(0) should not be used

File: property-services/src/main/java/org/egov/pt/service/PropertyService.java [hunk 3]
```diff
@@ -232,4 +236,281 @@ private Boolean enrichCriteriaFromUser(PropertyCriteria criteria, RequestInfo re
...
+			property.setId(UUID.randomUUID().toString());
+			property.setPropertyId(oldProperty.getPropertyId());
+			//property.setSurveyId();
+			property.setLinkedProperties(null);
+			property.setTenantId(oldProperty.getTenantId());
+			property.setAccountId(requestInfo.getUserInfo().getUuid());
+			property.setOldPropertyId(oldProperty.getOldPropertyId());
+			property.setStatus(Status.fromValue(oldProperty.getStatus().toString()));
+
+			if(oldProperty.getAddress() == null)
```
Comment (talele08) [inline property-services/src/main/java/org/egov/pt/service/PropertyService.java]: address will be always not null

File: property-services/src/main/java/org/egov/pt/service/PropertyService.java [hunk 4]
```diff
@@ -232,4 +236,281 @@ private Boolean enrichCriteriaFromUser(PropertyCriteria criteria, RequestInfo re
...
+			if(oldProperty.getPropertyDetails().get(0).getInstitution() == null)
+				property.setInstitution(null);
+			else
+				property.setInstitution(migrateInstitution(oldProperty.getPropertyDetails().get(0).getInstitution()));
+			property.setCreationReason(CreationReason.fromValue(String.valueOf(oldProperty.getCreationReason())));
+			property.setUsageCategory(migrateUsageCategory(oldProperty));
+			property.setNoOfFloors(oldProperty.getPropertyDetails().get(0).getNoOfFloors());
+			property.setLandArea(Double.valueOf(oldProperty.getPropertyDetails().get(0).getLandArea()));
+			property.setSuperBuiltUpArea(BigDecimal.valueOf(oldProperty.getPropertyDetails().get(0).getBuildUpArea()));
+			property.setSource(Source.fromValue(String.valueOf(oldProperty.getPropertyDetails().get(0).getSource())));
```
Comment (talele08) [inline property-services/src/main/java/org/egov/pt/service/PropertyService.java]: Will throw null pointer in valueOf

File: property-services/src/main/java/org/egov/pt/service/PropertyService.java [hunk 5]
```diff
@@ -232,4 +236,281 @@ private Boolean enrichCriteriaFromUser(PropertyCriteria criteria, RequestInfo re
...
+			property.setAcknowldgementNumber(oldProperty.getAcknowldgementNumber());
+			property.setPropertyType(oldProperty.getPropertyDetails().get(0).getPropertyType());
+			property.setOwnershipCategory(migrateOwnwershipCategory(oldProperty));
+			property.setOwners(migrateOwnerInfo(oldProperty.getPropertyDetails().get(0).getOwners()));
+			if(oldProperty.getPropertyDetails().get(0).getInstitution() == null)
+				property.setInstitution(null);
+			else
+				property.setInstitution(migrateInstitution(oldProperty.getPropertyDetails().get(0).getInstitution()));
+			property.setCreationReason(CreationReason.fromValue(String.valueOf(oldProperty.getCreationReason())));
+			property.setUsageCategory(migrateUsageCategory(oldProperty));
```
Comment (talele08) [inline property-services/src/main/java/org/egov/pt/service/PropertyService.java]: if condition on UsageCategory

File: property-services/src/main/java/org/egov/pt/service/PropertyService.java [hunk 6]
```diff
@@ -232,4 +236,281 @@ private Boolean enrichCriteriaFromUser(PropertyCriteria criteria, RequestInfo re
...
+			property.setSuperBuiltUpArea(BigDecimal.valueOf(oldProperty.getPropertyDetails().get(0).getBuildUpArea()));
+			property.setSource(Source.fromValue(String.valueOf(oldProperty.getPropertyDetails().get(0).getSource())));
+			property.setChannel(Channel.fromValue(String.valueOf(oldProperty.getPropertyDetails().get(0).getChannel())));
+
+			if(oldProperty.getPropertyDetails().get(0).getDocuments() == null)
+				property.setDocuments(null);
+			else
+				property.setDocuments(migrateDocument(oldProperty.getPropertyDetails().get(0).getDocuments()));
+
+			property.setUnits(migrateUnit(oldProperty.getPropertyDetails().get(0).getUnits()));
```
Comment (talele08) [inline property-services/src/main/java/org/egov/pt/service/PropertyService.java]: for vacant units are null will throw null pointer

File: property-services/src/main/java/org/egov/pt/service/PropertyService.java [hunk 7]
```diff
@@ -232,4 +236,281 @@ private Boolean enrichCriteriaFromUser(PropertyCriteria criteria, RequestInfo re
...
+			property.setAdditionalDetails(oldProperty.getPropertyDetails().get(0).getAdditionalDetails());
+			property.setAuditDetails(migrateAuditDetails(oldProperty.getOldAuditDetails()));
+			properties.add(property);
+
+			PropertyRequest request = PropertyRequest.builder().requestInfo(requestInfo).property(property).build();
+			//createProperty(request);
+
+			propertyValidator.validateCreateRequest(request);
+			if (config.getIsWorkflowEnabled())
```
Comment (talele08) [inline property-services/src/main/java/org/egov/pt/service/PropertyService.java]: workflow will be disabled for migration

File: property-services/src/main/java/org/egov/pt/service/PropertyService.java [hunk 8]
```diff
@@ -232,4 +236,281 @@ private Boolean enrichCriteriaFromUser(PropertyCriteria criteria, RequestInfo re
...
+				updateWorkflow(request, true);
+			producer.push(config.getSavePropertyTopic(), request);
```
Comment (talele08) [inline property-services/src/main/java/org/egov/pt/service/PropertyService.java]: create assessment object as well and call assessment validation on it

File: property-services/src/main/java/org/egov/pt/service/PropertyService.java [hunk 9]
```diff
@@ -232,4 +236,281 @@ private Boolean enrichCriteriaFromUser(PropertyCriteria criteria, RequestInfo re
...
+		locality.setLabel(oldLocality.getLabel());
+		locality.setLatitude(oldLocality.getLatitude());
+		locality.setLongitude(oldLocality.getLongitude());
+		locality.setArea(oldLocality.getArea());
+		locality.setMaterializedPath(oldLocality.getMaterializedPath());
+		locality.setChildren(setmigrateLocalityList(oldLocality.getChildren()));
+		return  locality;
+	}
+
+	public List<Locality> setmigrateLocalityList(List<Boundary> oldchildrenList){
```
Comment (talele08) [inline property-services/src/main/java/org/egov/pt/service/PropertyService.java]: not required

File: property-services/src/main/java/org/egov/pt/service/PropertyService.java [hunk 10]
```diff
@@ -232,4 +236,281 @@ private Boolean enrichCriteriaFromUser(PropertyCriteria criteria, RequestInfo re
...
+			geoLocation.setLatitude(Double.valueOf(oldAddress.getLatitude()));
+
+		if(oldAddress.getLongitude() == null)
+			geoLocation.setLongitude(null);
+		else
+			geoLocation.setLongitude(Double.valueOf(oldAddress.getLongitude()));
+		return  geoLocation;
+	}
+
+	public List<OwnerInfo> migrateOwnerInfo(Set<OldOwnerInfo> oldOwnerInfosSet){
```
Comment (talele08) [inline property-services/src/main/java/org/egov/pt/service/PropertyService.java]: not required to add user information as it is part of user service

File: property-services/src/main/java/org/egov/pt/service/PropertyService.java [hunk 11]
```diff
@@ -232,4 +236,281 @@ private Boolean enrichCriteriaFromUser(PropertyCriteria criteria, RequestInfo re
...
+		for(OldUnit oldUnit : oldUnits){
+			Unit unit = new Unit();
+			unit.setId(oldUnit.getId());
+			unit.setTenantId(oldUnit.getTenantId());
+			unit.setFloorNo(Integer.valueOf(oldUnit.getFloorNo()));
+			unit.setUnitType(oldUnit.getUnitType());
+			unit.setUsageCategory(migrateUnitUsageCategory(oldUnit));
+			unit.setOccupancyType(OccupancyType.fromValue(oldUnit.getOccupancyType()));
+			unit.setOccupancyDate(oldUnit.getOccupancyDate());
+			unit.setActive(Boolean.TRUE);
```
Comment (talele08) [inline property-services/src/main/java/org/egov/pt/service/PropertyService.java]: set it from oldUnit

**Comment by rohit-eGov** [comment]
Closing this PR as this Task is already done

**Pull request closed without merging**